### ISSUE TITLE
Lightgbm 2.1.2 different results

### ISSUE THREAD

**Issue by psinger** [Microsoft/LightGBM#1487]
I am currently trying out Lightgbm 2.1.2 and are getting quite different results compared to 2.1.0 fitting the same models with same hyperparameters (worse accuracy). 

Has there been any fundamental changes to the current version? Can't find any clear version notes.

**Comment by guolinke** [comment]
can you provide an reproduce code with random generated data?

**Comment by psinger** [comment]
@guolinke Unfortunately, I can't produce some random generated data right now. What I am doing is binary classification using LGBMClassifier python wrapper and features are highly dimensional based on bag of words. Evaluation is AUROC.

I will try to figure out some open dataset that leads to similar behavior and report back. Maybe you have some idea in the meantime.

**Comment by rgranvil** [comment]
@psinger You could try running with boost_from_average = false.  I noticed some slight changes in my binary classification benchmarks after updating to v2.1.2 caused by https://github.com/Microsoft/LightGBM/commit/441bafa182b08df765aa72f7e38eaf632eb93b10.  But for me the differences were very small with appropriate nrounds.

**Comment by psinger** [comment]
@rgranvil That was indeed it. Running it currently on low number of iterations (100) and AUROC changes from 0.88 to 0.93 by setting ```boost_from_average=False```.

**Comment by StrikerRUS** [comment]
Thanks @rgranvil !

**Comment by eryueniaobp** [comment]
also find the same issue.   2.1.2 will give a much weaker auc then version 2.1.0  (auc 0.73 to 0.79 in my case )   The data is highly imbalanced.

**Comment by psinger** [comment]
@eryueniaobp Have you checked `boost_from_average` as discussed above?
